key-uglifier/src/uglifier.rs

```diff
@@ -0,0 +1,107 @@
+use colored::Colorize;
+use hex::ToHex;
+
+fn param_set_default_encryption_key(key_length: u8) -> Vec<u8>
+{
+    let default_key = vec![0x62u8; key_length as usize];
+    return default_key
+}
+
+fn read_hex_nibble(c: char) -> (bool, String)
+{
+    if (c >='0') && (c <= '9'){
+        return (true, c.to_string());
+    }
+    else if (c >='A') && (c <= 'F')
+    {
+        return (true, c.to_string());
+    }
+    else if (c >='a') && (c <= 'f')
+    {
+        return (true, c.to_string());
+    }
+    else
+    {
+        // printf("[%u] read_hex_nibble: Error char not in supported range",nodeId);
+        return (false, '0'.to_string().on_color("green").to_string());
+    }
+}
+
+fn convert_to_hex(key_input: &String, key_length: u8) -> (bool, String) {
+    let mut key_output = String::new();
+    let mut bad_chars: bool = false;
```

Comment:
Types can be derived by the compiler in 90% of the cases. 
```suggestion
    let mut bad_chars = false;
```